src/lib/Liblog/pbs_log.c

```diff
@@ -908,7 +917,7 @@ log_record(int eventtype, int objclass, int sev, const char *objname, const char
 			log_err(rc, "log_record", "PBS cannot open its log");
 			fclose(logfile);
 		}
-		return;
+		goto sigunblock;
```

Comment:
This jump will be bad, if it happens, will leave the mutex locked given where you have it now

Reply:
We unlock the mutex at the beginning of this block (see line 913), so I think it should be fine.